src/NRules/NRules.Fluent/Dsl/IFilterExpression.cs

```diff
@@ -16,6 +17,16 @@ public interface IFilterExpression
         /// <returns>Filters expression builder.</returns>
         IFilterExpression OnChange(params Expression<Func<object>>[] keySelectors);
 
+        /// <summary>
+        /// Configures the engine to filter rule's matches, so that updates are only triggered if given key changed.
+        /// Comparison is done using the custom equality comparer given
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="keySelector">Key selector expression.</param>
+        /// <param name="comparer">Equality comparer for key</param>
+        /// <returns></returns>
+        IFilterExpression OnChange<T>(Expression<Func<T>> keySelector, Expression<Func<IEqualityComparer<T>>> comparer) where T : class;
```

Comment:
Why should the comparer be an expression? It seems sufficient to just pass the actual comparer instance. Or am I missing something?